<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1400</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>é‚®å±€é€‰å€é—®é¢˜ - æœ€çŸ­è·¯å¾„ç®—æ³•æ±‚è§£</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
/* å…¨å±€é€šç”¨æ ·å¼ */
QWidget {
    font-family: "Microsoft YaHei", "PingFang SC", sans-serif;
}
/* ç½‘æ ¼ç»˜å›¾åŒºåº•å±‚å®¹å™¨ - ä¸“é—¨æ‰¿è½½XYç½‘æ ¼ */
#gridContainer {
    background-color: #ffffff;
    border-radius: 16px;
    margin: 15px;
}
/* å·¦ä¾§é¢æ¿æ ·å¼ - æ‚¬æµ®æ„Ÿï¼Œä¸é®æŒ¡ç½‘æ ¼ */
#frame {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #f9fafb, stop:1 #f1f3f5);
    border-radius: 16px;
    border: none;
    box-shadow: 0 6px 16px rgba(0, 0, 0, 0.08);
    padding: 20px;
    /* é¢æ¿è½»å¾®æ‚¬æµ®ï¼Œä¸é®æŒ¡ç½‘æ ¼ */
    margin: 15px 0 15px 15px;
}
QLabel {
    color: #333333;
}
QPushButton {
    border-radius: 8px;
    font-weight: 600;
    transition: all 0.2s ease;
}
QPushButton:hover {
    transform: translateY(-2px);
}
QPushButton:pressed {
    transform: translateY(0);
}
   </string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <!-- åº•å±‚ç½‘æ ¼å®¹å™¨ï¼šæ‰¿è½½XYç½‘æ ¼ï¼Œé“ºæ»¡æ•´ä¸ªcentralwidget -->
   <widget class="QWidget" name="gridContainer">
    <layout class="QHBoxLayout" name="rootLayout">
     <property name="spacing">
      <number>0</number> <!-- æ¶ˆé™¤å¸ƒå±€é—´è·ï¼Œé¿å…ç½‘æ ¼è¢«åˆ‡å‰² -->
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <!-- å·¦ä¾§é¢æ¿ï¼šå›ºå®šå®½åº¦ï¼Œä¸æŒ¤å‹ç½‘æ ¼åŒº -->
     <item>
      <widget class="QFrame" name="frame">
       <property name="minimumSize">
        <size>
         <width>320</width>
         <height>700</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>350</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::Shape::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Shadow::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>15</number>
        </property>
        <property name="leftMargin">
         <number>5</number>
        </property>
        <property name="topMargin">
         <number>5</number>
        </property>
        <property name="rightMargin">
         <number>5</number>
        </property>
        <property name="bottomMargin">
         <number>5</number>
        </property>
        <item>
         <widget class="QLabel" name="label">
          <property name="styleSheet">
           <string notr="true">
font-size: 26px;
font-weight: 700;
color: white;
background: qlineargradient(x1:0, y1:0, x2:1, y2:0, stop:0 #4361ee, stop:1 #3a0ca3);
padding: 20px;
border-radius: 12px;
margin-bottom: 10px;
box-shadow: 0 4px 8px rgba(67, 97, 238, 0.2);
text-align: center;
           </string>
          </property>
          <property name="text">
           <string>ğŸ¤ é‚®å±€é€‰å€é—®é¢˜</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="styleSheet">
           <string notr="true">
font-size: 18px;
font-weight: 600;
color: #2d3748;
background-color: #edf2f7;
padding: 12px 16px;
border-radius: 8px;
border-left: 4px solid #4361ee;
           </string>
          </property>
          <property name="text">
           <string>ğŸ“‹ æ“ä½œè¯´æ˜</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="styleSheet">
           <string notr="true">
font-size: 14px;
color: #4a5568;
background-color: white;
padding: 16px;
border-radius: 8px;
border: 1px solid #e2e8f0;
line-height: 1.8;
box-shadow: 0 2px 4px rgba(0, 0, 0, 0.03);
           </string>
          </property>
          <property name="text">
           <string>â€¢ ğŸ–±ï¸ ç‚¹å‡»å³ä¾§ç»˜å›¾åŒºåŸŸæ·»åŠ å±…æ°‘å°åŒº
â€¢ âš¡ ä½¿ç”¨ä¸‹æ–¹æŒ‰é’®è¿›è¡Œç›¸å…³æ“ä½œ
â€¢ ğŸ“Š ç³»ç»Ÿè‡ªåŠ¨è®¡ç®—æœ€ä¼˜é‚®å±€ä½ç½®
â€¢ ğŸ”„ æ”¯æŒæ¸…ç©ºå’Œéšæœºç”ŸæˆåŠŸèƒ½
â€¢ ğŸ“ ä½¿ç”¨æ›¼å“ˆé¡¿è·ç¦»ä¼˜åŒ–ç®—æ³•</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="btnCalculate">
          <property name="styleSheet">
           <string notr="true">
QPushButton {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #4361ee, stop:1 #3a0ca3);
    color: white;
    border: none;
    padding: 16px 24px;
    font-size: 16px;
    min-height: 50px;
    box-shadow: 0 4px 8px rgba(67, 97, 238, 0.2);
}
QPushButton:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #3a0ca3, stop:1 #2a0878);
    box-shadow: 0 6px 12px rgba(67, 97, 238, 0.3);
}
QPushButton:pressed {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #2a0878, stop:1 #1e0659);
}
           </string>
          </property>
          <property name="text">
           <string>ğŸ“Š è®¡ç®—æœ€ä¼˜é‚®å±€ä½ç½®</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnClear">
          <property name="styleSheet">
           <string notr="true">
QPushButton {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #ef4444, stop:1 #dc2626);
    color: white;
    border: none;
    padding: 16px 24px;
    font-size: 16px;
    min-height: 50px;
    box-shadow: 0 4px 8px rgba(239, 68, 68, 0.2);
}
QPushButton:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #dc2626, stop:1 #b91c1c);
    box-shadow: 0 6px 12px rgba(239, 68, 68, 0.3);
}
QPushButton:pressed {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #b91c1c, stop:1 #991b1b);
}
           </string>
          </property>
          <property name="text">
           <string>ğŸ—‘ï¸ æ¸…ç©ºæ‰€æœ‰å°åŒº</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnRandom">
          <property name="styleSheet">
           <string notr="true">
QPushButton {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #10b981, stop:1 #059669);
    color: white;
    border: none;
    padding: 16px 24px;
    font-size: 16px;
    min-height: 50px;
    box-shadow: 0 4px 8px rgba(16, 185, 129, 0.2);
}
QPushButton:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #059669, stop:1 #047857);
    box-shadow: 0 6px 12px rgba(16, 185, 129, 0.3);
}
QPushButton:pressed {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #047857, stop:1 #065f46);
}
           </string>
          </property>
          <property name="text">
           <string>ğŸ² éšæœºç”Ÿæˆå°åŒº</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="styleSheet">
           <string notr="true">
font-size: 18px;
font-weight: 600;
color: #2d3748;
background-color: #edf2f7;
padding: 12px 16px;
border-radius: 8px;
border-left: 4px solid #4361ee;
           </string>
          </property>
          <property name="text">
           <string>ğŸ§® ç®—æ³•ä¿¡æ¯</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblAlgorithmInfo">
          <property name="styleSheet">
           <string notr="true">
font-size: 14px;
color: #4a5568;
background-color: white;
padding: 16px;
border-radius: 8px;
border: 1px solid #e2e8f0;
line-height: 1.8;
box-shadow: 0 2px 4px rgba(0, 0, 0, 0.03);
           </string>
          </property>
          <property name="text">
           <string>â€¢ ğŸ“Š ä½¿ç”¨ä¸­ä½æ•°æ–¹æ³•æ±‚è§£
â€¢ ğŸ“ æ›¼å“ˆé¡¿è·ç¦»æœ€å°åŒ–
â€¢ âš¡ æ—¶é—´å¤æ‚åº¦ï¼šO(n)
â€¢ âœ… ä¿è¯æœ€ä¼˜è§£
â€¢ ğŸ¯ æ”¯æŒåŠ¨æ€æ·»åŠ å°åŒº</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <!-- å³ä¾§ç½‘æ ¼ç»˜å›¾åŒºï¼šè‡ªåŠ¨å¡«å……å‰©ä½™ç©ºé—´ï¼Œæ‰¿è½½XYç½‘æ ¼ -->
     <item>
      <widget class="QLabel" name="label_5">
       <property name="styleSheet">
        <string notr="true">
background-color: transparent;
/* ç»˜å›¾åŒºå†…è¾¹è·ï¼Œé¿å…ç½‘æ ¼è´´è¾¹ */
padding: 20px;
        </string>
       </property>
       <property name="text">
        <string/>
       </property>
       <!-- å…è®¸ç»˜å›¾åŒºæ‰©å±•ï¼Œé“ºæ»¡å³ä¾§ -->
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>1</horstretch>
         <verstretch>1</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1400</width>
     <height>18</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">
background-color: #f8fafc;
border: none;
    </string>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="styleSheet">
    <string notr="true">
background-color: #f8fafc;
color: #4a5568;
border-top: 1px solid #e2e8f0;
    </string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>